src/legacy/config/config-set.ts

```diff
@@ -218,6 +216,9 @@ export class ConfigSet {
     this._matchTestFilePath = globsToMatcher(
       this._matchablePatterns.filter((pattern: string | RegExp) => typeof pattern === 'string') as string[],
     )
+
+    this._setupConfigSet(options)
```

Comment:
`_setupConfigSet` now depends on `matchablePatterns` to exist when its called, because `isTestFile` is used in finding the appropriate tsconfig reference project to use for ts compile/transpile process.

Reply:
If we only check for test files, would it be like exceptional case when processing a random file? Should we also treat any files given through `process` method in transformer class similarly like this implementation?

Reply:
When processing a random file, it defaults to the current behavior - finds a relevant tsconfig.json to pick, as it currently does.

Reply:
Does it mean the handling with project reference config only applies to test files as you mentioned the file name needs to pass the check `isTestFile`?

Reply:
Yes, the rationale is - if there is a specific tsconfig project reference that addresses jest files alone, we should use that tsconfig file. Otherwise we default to current behavior of using the root tsconfig.json file we found.

Reply:
I wonder if this can break any existing behaviors? In theory, it should fix instead of breaking.

In reality, this actually changes the existing behavior into a different behavior. Do you think that is a breaking change since we changed the behavior? I still suspect there might be edge cases where this change can cause a breaking change.

Reply:
All LGTM 👍

One final request is: would you please edit https://github.com/kulshekhar/ts-jest/tree/main/examples/monorepo-app to use project references to demo this change? That would be a nice example to show to others that it actually works

Reply:
I will!

Reply:
@ahnpnl here is the commit with the changes to example repo. Please check it out - https://github.com/kulshekhar/ts-jest/pull/4689/commits/d19aee323d332b32aab46a90dec444f7baf93bac

and let me know if it looks okay and approve if you think so.

Thank you.